IdentityCore/src/workplacejoin/MSIDPkeyAuthHelper.m

```diff
@@ -76,6 +77,9 @@ + (nullable NSString *)createDeviceAuthResponse:(nonnull NSURL *)authorizationSe
         {
             authToken = [NSString stringWithFormat:@"AuthToken=\"%@\",", [MSIDPkeyAuthHelper createDeviceAuthResponse:authorizationServerComponents.string nonce:[challengeData valueForKey:@"nonce"] identity:info]];
             MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"Found WPJ Info and responded to PKeyAuth Request.");
+            // Save telemetry for successful PkeyAuth ADFS challenge responses
+            NSUInteger httpStatusCode = [challengeData valueForKey:@"SubmitUrl"] ? 302 : 401;
```

Comment:
Can you clarify why presence of SubmitURL is important for this?

Reply:
Yes, according to Feature requirements table row 1.5 in [Device Authentication using PkeyAuth](https://microsoft.sharepoint.com/:w:/t/aad/devex/EfksUgoBpPNGnACpeQ-_puwBj5JZ616HHxQYxvKJm3JTMg?e=H6BbHm), submitUrl will be present in challengeData for interactive flows.
302 is the Http response status for interactive flows and 401 is the Http response status for non-interactive flows.
I'm using this to determine what the status code would be and include it in the telemetry. (Will give us insight about if challenge was interactive / non-interactive)

The telemetry will look like : 4|0|ADFS_PKEYAUTH_CHLG_fs.msidlab4.com_302|

Reply:
Thanks, looking good to me.